File: packages/typescript/src/resources/sandbox.ts
/**
 * The `sandbox` resource: rendered text of sandbox sends, including OTP codes.
 * @module
 */
import type { ListParams, Page, RequestOptions, SandboxMessage } from '../models.js';
import { Resource, wireQuery } from './base.js';

/** Accessed as `client.sandbox`. */
export class Sandbox extends Resource {
  listMessages(params: ListParams = {}, options?: RequestOptions): Promise<Page<SandboxMessage>> {
    return this.page({ method: 'GET', path: '/v1/sandbox/messages', query: wireQuery(params) }, options);
  }
}

File: packages/rust/src/resources/sandbox.rs
//! The `sandbox` resource: rendered sandbox messages (including OTP codes).

use super::list_query;
use crate::error::Result;
use crate::http::HttpClient;
use crate::models::{ListParams, Page, SandboxMessage};

/// Accessed as `client.sandbox()`.
#[derive(Debug, Clone)]
pub struct Sandbox {
    http: HttpClient,
}

impl Sandbox {
    pub(crate) fn new(http: HttpClient) -> Self {
        Self { http }
    }

    /// List sandbox messages with their rendered text.
    pub async fn list_messages(&self, params: ListParams) -> Result<Page<SandboxMessage>> {
        self.http
            .get(&format!("/v1/sandbox/messages{}", list_query(&params)))
            .await
    }
}
